pkg/distrosort/sort_test.go

```diff
@@ -0,0 +1,58 @@
+package distrosort_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/osbuild/images/pkg/distrosort"
+)
+
+func TestSortNames(t *testing.T) {
+	for _, tc := range []struct {
+		inp    []string
+		sorted []string
+	}{
+		{
+			// distro names are sorted by first
+			[]string{"foo-2", "bar-1", "foo-1"},
+			[]string{"bar-1", "foo-1", "foo-2"},
+		}, {
+			// multiple "-" are okay
+			[]string{"foo-bar-2", "bar-foo-1", "foo-bar-1"},
+			[]string{"bar-foo-1", "foo-bar-1", "foo-bar-2"},
+		}, {
+			// 1.4 is smaller than 1.10, sort.Strings will get this
+			// wrong
+			[]string{"foo-1.10", "foo-1.4"},
+			[]string{"foo-1.4", "foo-1.10"},
+		},
+	} {
+		err := distrosort.Names(tc.inp)
+		assert.NoError(t, err, tc.inp)
+		assert.Equal(t, tc.sorted, tc.inp)
+	}
+}
+
+func TestSortNamesInvalidVersion(t *testing.T) {
+	for _, tc := range []struct {
+		inp         []string
+		expectedErr string
+	}{
+		{
+			[]string{"foo-1.x", "foo-2"},
+			"error when parsing distro name (foo-1.x): parsing minor version failed, inner error:\nstrconv.Atoi: parsing \"x\": invalid syntax",
+		}, {
+			// missing "-" is not supported
+			[]string{"foo", "bar-1"},
+			`error when parsing distro name (foo): A dash is expected to separate distro name and version`,
+		}, {
+			// foo-1.4-beta is not supported
+			[]string{"foo-1.4", "foo-1.4-beta", "foo-1.0"},
+			"error when parsing distro name (foo-1.4-beta): parsing major version failed, inner error:\nstrconv.Atoi: parsing \"beta\": invalid syntax",
```

Comment:
Is it correct that this is failing on the *major* here? It seems to me that if we don't support the `-beta` part that it would be part of the minor version?

Reply:
The parser is currently quite simple, it will assume the major/minor after the first or second "-" (second to support `centos-stream-9`. I will investigate if we can easily fix this without breaking compatbility

Reply:
I would also point out that each distro may have its own parser implementation (e.g., an RHEL distro before RHEL-10 to support release numbers without a dot).

Reply:
In that case feel free to resolve this :)